stdlib/ast.pyi

```diff
@@ -34,6 +34,12 @@ class _Attributes(TypedDict, Generic[_EndPositionT], total=False):
 # but they consider themselves to live in the ast module,
 # so we'll define the stubs in this file.
 class AST:
+    @overload
+    def __init__(self) -> None: ...
+    if sys.version_info < (3, 15):
+        @overload
+        @deprecated("Support for arbitrary keyword arguments is deprecated and will be removed in Python 3.15.")
+        def __init__(self, **kwargs: Any) -> None: ...
```

Comment:
I also considered the following (see https://github.com/python/typeshed/issues/4760#issuecomment-2791428063):
```suggestion
class AST(Generic[_CustomASTArgumentT]):
    @overload
    def __init__(self) -> None: ...
    if sys.version_info < (3, 15):
        @overload
        @deprecated("Support for arbitrary keyword arguments is deprecated and will be removed in Python 3.15.")
        def __init__(self, **kwargs: _CustomASTArgumentT) -> None: ...
        def __getattribute__(self, name: str) -> _CustomASTArgumentT: ...
```